src/Acuminator/Acuminator.Analyzers/Analyzers/Localization/LocalizationPXExceptionAnalyzer.cs

```diff
@@ -0,0 +1,110 @@
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics =>
+            ImmutableArray.Create
+            (
+                Descriptors.PX1050_HardcodedStringInLocalizationMethod,
+                Descriptors.PX1051_NonLocalizableString,
+                Descriptors.PX1053_ConcatinationPriorLocalization
+            );
+
+        internal override void AnalyzeCompilation(CompilationStartAnalysisContext compilationStartContext, PXContext pxContext)
+        {
+            compilationStartContext.RegisterSyntaxNodeAction(syntaxContext => AnalyzeThrowOfPXException(syntaxContext, pxContext), SyntaxKind.ThrowStatement);
+            compilationStartContext.RegisterSyntaxNodeAction(syntaxContext => AnalyzePXExceptionCtorInitializer(syntaxContext, pxContext), SyntaxKind.ClassDeclaration);
+        }
+
+        private void AnalyzePXExceptionCtorInitializer(SyntaxNodeAnalysisContext syntaxContext, PXContext pxContext)
+        {
+            if (!(syntaxContext.Node is ClassDeclarationSyntax classDeclaration) || syntaxContext.CancellationToken.IsCancellationRequested)
+                return;
+
+            INamedTypeSymbol type = syntaxContext.SemanticModel.GetDeclaredSymbol(classDeclaration);
+            bool isPXException = type != null && type.InheritsFromOrEquals(pxContext.PXException);
+
+            if (!isPXException)
+                return;
+
+            IEnumerable<ConstructorInitializerSyntax> baseCtors = classDeclaration.DescendantNodes()
+                                                                  .OfType<ConstructorInitializerSyntax>();
+            foreach (ConstructorInitializerSyntax c in baseCtors)
+            {
+                if (!(syntaxContext.SemanticModel.GetSymbolInfo(c, syntaxContext.CancellationToken).Symbol is IMethodSymbol methodSymbol))
+                    continue;
+
+                ImmutableArray<IParameterSymbol> pars = methodSymbol.Parameters;
+                ExpressionSyntax messageExpression = GetMessageExpression(syntaxContext, pars, c.ArgumentList);
+
+                if (messageExpression == null)
+                    continue;
+
+                LocalizationMessageHelper messageHelper = new LocalizationMessageHelper(syntaxContext, pxContext, messageExpression, false);
+                messageHelper.ValidateMessage();
+            }
+        }
+
+        private ExpressionSyntax GetMessageExpression(SyntaxNodeAnalysisContext syntaxContext, ImmutableArray<IParameterSymbol> pars, ArgumentListSyntax args)
+        {
+            if (syntaxContext.CancellationToken.IsCancellationRequested)
+                return null;
+
+            ArgumentSyntax messageArg = null;
+
+            foreach (ArgumentSyntax a in args.Arguments)
+            {
+                IParameterSymbol p = a.DetermineParameter(pars, false, syntaxContext.CancellationToken);
+
+                if (p != null && !string.IsNullOrEmpty(p.Name) && _messageArgNames.Contains(p.Name, StringComparer.Ordinal))
```

Comment:
@metallmonster just `_messageArgNames.Contains(p?.Name, StringComparer.Ordinal)` is enough since it handles null

Reply:
@SENya1990 I agree